### PR TITLE
JS program execution

### LINKED ISSUE #259: Program Execution in JavaScript
Will document the execution context, its phases and call stack
Can I work on this issue?

### PR DESCRIPTION
**Pull request by aakankshabhende**
<hr>

## Description 📜
Created a JavaScript folder and added documentation for Js program execution and call stack.
Fixes #259  

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] JavaScript

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

### CODE REVIEW

File: JavaScript/JS-Program-Execution.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+
+# Program Execution in JavaScript
+
+When we run a JS program, an execution context is created, which is further divided into two phases:
+        
+        1.  Memory Allocation Phase (Variable Environment)
+        2.  Code Execution Phase (Thread of Execution)
+
+## First Phase
+In the first phase, the entire program is skimmed line by line and memory is allocated to functions and variables. In case of variables, undefined is stored as a placeholder and in case of functions, the whole code of function is stored.
```
Comment (s4-2) [inline JavaScript/JS-Program-Execution.md:10]: - In the first phase, the entire program is skimmed line by line, and memory is allocated to functions and variables. In the case of variables, undefined is stored as a placeholder and in the case of functions, the whole code of the function is stored.
Some grammatical and punctuation mistakes need to be corrected

File: JavaScript/JS-Program-Execution.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+## Second Phase
+In the second phase, the code is actually executed line by line. Undefined keyword is now replaced by actual values of variables. When there is a function invocation, a entirely new execution context is created which has further two components. When return keyword is encountered, it just returns all the controls to the global execution context where the function was first invoked. After complete execution of function, the execution context related to the instance of function is deleted. Then, the next line is executed.
```
Comment (s4-2) [inline JavaScript/JS-Program-Execution.md:13]: - In the second phase, the code is actually executed line by line. **The** Undefined keyword is now replaced by actual values of variables. When there is a function invocation, a entirely new execution context is created which has further two components. When return keyword is encountered, it just returns all the controls to the global execution context where the function was first invoked. After complete execution of **the** function, the execution context related to the instance of **the**  function is deleted. Then, the next line is executed.

File: JavaScript/JS-Program-Execution.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+## Call Stack
+When a program starts execution JavaScript pushes the whole program as global context into a stack which is known as Call Stack and continues execution. Whenever JavaScript executes a new context and just follows the same process and pushes to the stack.
```
Comment (s4-2) [inline JavaScript/JS-Program-Execution.md:16]: -  When a program starts execution JavaScript pushes the whole program as global context into a stack which is known as Call Stack and continues execution. Whenever JavaScript executes a new context and just follows the same process and pushes to the stack.

**Comment by s4-2** [comment]
Hey @aakankshabhende , Good job. Just fix some grammatical errors and you are good to go:). Feel free to ask any queries.

**Comment by aakankshabhende** [comment]
Done @s4-2 mam

**Comment by s4-2** [comment]
@aakankshabhende Great. Add the hyphen before every sentence to make it a point. I think it will look good.

**Comment by aakankshabhende** [comment]
Done @s4-2 mam

**Comment by s4-2** [comment]
Hi @aakankshabhende , hyphen means ( - ) this.
Hope you got it now:)

**Comment by aakankshabhende** [comment]
@s4-2 mam can you please guide how to do that in markdown language

**Comment by s4-2** [comment]
@aakankshabhende Sure:)
Just add this ' - ' before starting of the sentence.
eg : - GirlScript Winter of Contribution.
Please let me know if you understood?

**Comment by aakankshabhende** [comment]
Mam it adds a bullet point

**Comment by aakankshabhende** [comment]
Done @s4-2 mam

**Comment by s4-2** [comment]
Yes.

**Comment by s4-2** [comment]
Please ask for review in the respective channel, after successful review by two mentors your file can be merge.

**Comment by s4-2** [comment]
Please add your name at the bottom of the doc. Refer someone other PRs for it so you will understand better.

**Comment by paramsiddharth** [comment]
@aakankshabhende Really good contribution. However, I have to close the PR because of the huge diff created. You didn't branch out of the `Javascript` branch. Instead, you created a new branch from the `main` branch itself. This caused commits from other branches to appear in this PR.

Also, use `JavaScript/Program-Execution/README.md` instead of `JavaScript/JS-Program-Execution.md`.

Please preserve your contribution, create a fresh branch and a fresh PR. Make sure the branch is created out of `Javascript`, not `main`.

**Pull request closed without merging**